Src/WitsmlExplorer.Api/Services/LogDataReader.cs

```diff
@@ -166,14 +166,18 @@ public async Task<WitsmlLogData> GetNextBatch()
 
             try
             {
-                return await _buffer.ReceiveAsync(_receiveTokenSource.Token);
+                return await _buffer.ReceiveAsync(cancellationToken != null ? cancellationToken.Value : _receiveTokenSource.Token);
```

Comment:
This completely ignores _receiveTokenSource if canncelationToken is set. This can cause problems in the `Produce` method where `_receiveTokenSource` is used to cancel the if the buffer is empty or we got an exception.

Haven't tested, but maybe `CancellationTokenSource.CreateLinkedTokenSource` works for this scenario?
Something like
```
using var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(_receiveTokenSource.Token, cancellationToken);
return await _buffer.ReceiveAsync(linkedCts.Token);
```

Reply:
fixed, use linked token only when cancallationToken is not null